### PR TITLE
fix: add reserved port for vnet if custom node config port range exceeds 65330

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**

**What this PR does / why we need it**:
https://learn.microsoft.com/en-us/azure/virtual-network/virtual-networks-faq#what-protocols-can-i-use-within-vnets

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/scenario/scenario_marinerv2-custom-sysctls.go [hunk 1]
```diff
@@ -8,10 +8,11 @@ import (
 
 func marinerv2CustomSysctls() *Scenario {
 	customSysctls := map[string]string{
-		"net.ipv4.ip_local_port_range":       "32768 65535",
+		"net.ipv4.ip_local_port_range":       "32768 62535",
```
Comment (cameronmeissner) [inline e2e/scenario/scenario_marinerv2-custom-sysctls.go:11]: lol I think that fix I made a while ago to change the sshkey file name actually worked, this just reminded me

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -1068,6 +1080,10 @@ net.ipv4.tcp_tw_reuse={{if $s.NetIpv4TcpTwReuse}}1{{else}}0{{end}}
 {{- end}}
 {{- if $s.NetIpv4IpLocalPortRange}}
 net.ipv4.ip_local_port_range={{$s.NetIpv4IpLocalPortRange}}
+{{$rangeEnd := getPortRangeEndValue $s.NetIpv4IpLocalPortRange " " 1}}
+{{ if ge $rangeEnd 65330}}
+net.ipv4.ip_local_reserved_ports=65330
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:1085]: can we also add something to baker_test.go for this?

File: pkg/agent/baker.go [hunk 3]
```diff
@@ -945,6 +948,15 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 	}
 }
 
+func getPortRangeEndValue(str, delimiter string, index int) int {
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:951]: can the port range ever be anything else than "A B"? if not then we probably don't need delimiter/index args here

Comment (UtheMan) [reply pkg/agent/baker.go:951]: Good point, added it at first cause I tried to make it more generic, in the end it ended up like this :D

Comment (cameronmeissner) [reply pkg/agent/baker.go:951]: maybe also worth trimming from the left and right side as well, though not sure if that would ever actually help

File: pkg/agent/baker.go [hunk 4]
```diff
@@ -928,7 +929,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			return config.SSHStatus == datamodel.SSHOff
 		},
 		"GetSysctlContent": func() (string, error) {
-			sysctlTemplate, err := template.New("sysctl").Parse(sysctlTemplateString)
+			templateFuncMap := make(template.FuncMap)
+			templateFuncMap["getPortRangeEndValue"] = getPortRangeEndValue
```
Comment (alexeldeib) [inline pkg/agent/baker.go:933]: why define this as a template func and not call the go func directly? we do this in many places for e.g. gpu driver version, you don't need extra indirection -- that's what the map of template funcs already provides.

Comment (alexeldeib) [reply pkg/agent/baker.go:933]: oh jk, it's not the same template

**Review by alexeldeib — APPROVED** [review]

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**